### DISCUSSION TITLE
A consumer subscribing to more than one topics. Can it have separate process methods?

### DISCUSSION THREAD

**Discussion by rohitkumarrana67** [zendesk/racecar#282] [General] [answered]
class TestConsumer< Racecar::Consumer
  subscribes_to "comments", start_from_beginning: false
  subscribes_to "events", start_from_beginning: false

  def process(message)
    p "Received message: #{message.value}"
  end
end

**Comment by dasch** [comment, marked as answer]
You can easily set that up yourself:

```ruby
def process(message)
  send("process_#{message.topic}", message)
end

private

def process_comments(message)
  # ...
end

def process_events(message)
  # ...
end
``
